Feature/rename branch menu

mwerle opened the pull request:
Adds a right-click menu item to the Commit List to rename branches.
Also adds a menu item to rename the current branch in the main menu's "Branch" submenu.

Yes, this functionality is already available in the "MENU->Branch->Configure Branches" dialog, but this is something I semi-regularly use and it actually took me some time to discover the functionality in the dialog, especially since clicking on the branch name does not work reliably to edit it.

I feel that by adding it to the right-click menu it is much more discoverable.

I also refactored  the menu so that the "Delete Branch", "Delete Tag", and "Checkout" menu items turn into sub-menus when there are several references on a commit, making the menu more manageable. Again, this is something I encounter a bit in the repos I work with.

Murmele commented on src/ui/CommitList.cpp:
```diff
@@ -1573,19 +1606,23 @@ void CommitList::contextMenuEvent(QContextMenuEvent *event) {
 
       git::Reference head = view->repo().head();
-      foreach (const git::Reference &ref, commit.refs()) {
+      auto submenu = &menu;
+      auto entryName = tr("Checkout %1");
+      if (all_branches.count() > 1) {
+        submenu = menu.addMenu(tr("Checkout"));
+        entryName = QString("%1");
+      }
+      for (const git::Reference &ref : all_branches) {
```
Can you use also here the addMenuEntries function?

mwerle replied:
Unfortunately no because the code which actually generates the menu entries does more work specific to the checkout entries, such as enabling/disabling the menu entry, adding a tooltip, and, in the case of remote branches, calculating a local branch name for it.

Murmele replied:
ok I see thanks

Murmele commented:
which version of clang-format are you using to format the code, Version 13 is required, otherwise it makes some problems

mwerle commented:
Whichever is the default on my dev pc. At home it's v14.

Apologies if this version requirement is documented, I'll see if i can get v13 of clang-format.

Interesting if a different version of clang-format produces different results for the same configuration though..

Also if Gittyup requires a very specific build/verification environment, perhaps setting up a docker based build environment might be useful?

Murmele commented:
> Also if Gittyup requires a very specific build/verification environment, perhaps setting up a docker based build environment might be useful?

Yes this is a good idea. Just revert all changes in the wrong formatted file and then it should work again. For Arch linux clang-format 13 binaries are available

mwerle commented:
> > Also if Gittyup requires a very specific build/verification environment, perhaps setting up a docker based build environment might be useful?
> 
> Yes this is a good idea. Just revert all changes in the wrong formatted file and then it should work again. For Arch linux clang-format 13 binaries are available

So I did all the clean up and adding optional git hook in my other PR (#618) - if you are happy with my changes there, I'll merge them into this PR as well and/or create a separate PR just to fix the clang-format issue separately. As you prefer :)

mwerle commented:
Fixed the formatting, cherry-picked my change to force usage of clang-format v13 from my other PR, updated `changelog.md`, and rebased on the current `master` branch.

Murmele commented:
I will fix the linux build error whit this: https://github.com/Murmele/Gittyup/pull/627

Murmele commented:
Can you rebase on master? I fixed the changelog problem there

mwerle commented:
> Can you rebase on master? I fixed the changelog problem there

Huh.. I thought I already rebased on `master` before my last push..  done.

Murmele commented:
> 

Yes, I had to fix something there, otherwise the CI would not run through

Murmele commented:
@mwerle Thanks for the feature!
